Sources/HaCWebsiteLib/Models/NewWorkshop+init.swift

```diff
@@ -0,0 +1,285 @@
+import Foundation
+import Yaml
+
+let filePaths = (
+  metadata: "/info/metadata.yaml",
+  description: "/info/description.md",
+  prerequisites: "/info/prerequisites.md",
+  promoImagesDirectory: "/info/promo_images",
+  notesDirectory: "/content/notes",
+  notesMarkdown: "/content/notes/notes.md",
+  examplesDirectory: "/examples",
+  presenterGuide: "/content/presenter_guide.md",
+  setupInstructions: "/info/setup_instructions.md"
+)
+
+let validImageExtensions = ["jpg", "svg", "png", "gif"]
+
+extension NewWorkshop {
+  enum Error: Swift.Error {
+    case invalidPath
+    case missingMetadata
+    case malformedMetadata(String)
+    case missingDescription
+    case missingPrerequisites
+    case missingPromoImageBackground
+    case missingPromoImageForeground
+    case multiplePromoImageForegrounds
+    case multiplePromoImageBackgrounds
+    case invalidPromoImageForegroundFormat
+    case invalidPromoImageBackgroundFormat
+    case missingNotes
+    case badURLInNotes(String)
+    case emptyExamples
+    case invalidURL(String)
+  }
+
+  /// Create a NewWorkshop from a local path
+  init(localPath: String) throws {
+    workshopId = try NewWorkshop.getWorkshopId(localPath: localPath)
+    let remoteRepoUrl = "https://github.com/hackersatcambridge/workshop-" + workshopId
+
+    // TODO: Save the presenterGuide.
+    // This is currently run but not saved so that it still runs checks on the file
+    _ = try NewWorkshop.getPresenterGuide(localPath: localPath)
```

Comment:
Why are we not saving this?

Reply:
Good question. The format of this guide hadn’t been discussed very much so I didn’t save it but it’s no extra work to do so. So I will!